lua/astrocommunity/editing-support/codecompanion-nvim/init.lua

```diff
@@ -0,0 +1,156 @@
+                refresh_every_n_prompts = 0, -- e.g., 3 to refresh after every 3rd user prompt
+                ---Maximum number of times to refresh the title (default: 3)
+                max_refreshes = 3,
+              },
+              ---On exiting and entering neovim, loads the last chat on opening chat
+              continue_last_chat = false,
+              ---When chat is cleared with `gx` delete the chat from history
+              delete_on_clearing_chat = false,
+              ---Directory path to save the chats
+              dir_to_save = vim.fn.stdpath "data" .. "/codecompanion-history",
+              ---Enable detailed logging for history extension
+              enable_logging = false,
+              ---Optional filter function to control which chats are shown when browsing
+              chat_filter = nil, -- function(chat_data) return boolean end
+            },
+          },
+        },
+      },
+    },
+    { "AstroNvim/astroui", opts = { icons = { CodeCompanion = "󱙺" } } },
+    {
+      "AstroNvim/astrocore",
+      opts = function(_, opts)
+        if not opts.mappings then opts.mappings = {} end
+        opts.mappings.n = opts.mappings.n or {}
+        opts.mappings.v = opts.mappings.v or {}
+        opts.mappings.n[prefix] = { desc = require("astroui").get_icon("CodeCompanion", 1, true) .. "CodeCompanion" }
+        opts.mappings.v[prefix] = { desc = require("astroui").get_icon("CodeCompanion", 1, true) .. "CodeCompanion" }
+        opts.mappings.n[prefix .. "c"] = { "<cmd>CodeCompanionChat Toggle<cr>", desc = "Toggle chat" }
+        opts.mappings.v[prefix .. "c"] = { "<cmd>CodeCompanionChat Toggle<cr>", desc = "Toggle chat" }
+        opts.mappings.n[prefix .. "p"] = { "<cmd>CodeCompanionActions<cr>", desc = "Open action palette" }
+        opts.mappings.v[prefix .. "p"] = { "<cmd>CodeCompanionActions<cr>", desc = "Open action palette" }
+        opts.mappings.n[prefix .. "q"] = { "<cmd>CodeCompanion<cr>", desc = "Open inline assistant" }
+        opts.mappings.v[prefix .. "q"] = { "<cmd>CodeCompanion<cr>", desc = "Open inline assistant" }
+        opts.mappings.v[prefix .. "a"] = { "<cmd>CodeCompanionChat Add<cr>", desc = "Add selection to chat" }
+        vim.cmd [[cab cc CodeCompanion]]
+      end,
+    },
+    {
+      "MeanderingProgrammer/render-markdown.nvim",
+      optional = true,
+      opts = function(_, opts)
+        if not opts.file_types then opts.file_types = { "markdown" } end
+        opts.file_types = require("astrocore").list_insert_unique(opts.file_types, { "codecompanion" })
+      end,
+    },
+    {
+      "OXY2DEV/markview.nvim",
+      optional = true,
+      opts = function(_, opts)
+        if not opts.preview then opts.preview = {} end
+        if not opts.preview.filetypes then opts.preview.filetypes = { "markdown", "quarto", "rmd" } end
+        opts.preview.filetypes = require("astrocore").list_insert_unique(opts.preview.filetypes, { "codecompanion" })
+      end,
+    },
+    {
```

Comment:
Probably not needed. I think it internally picks whichever user has: https://codecompanion.olimorris.dev/configuration/action-palette.html